### PR TITLE
Provide expando to show the fingerprint of a key

### PR DESCRIPTION
**Pull request by rayfordshire**
* **What does this PR do?**

1) The first commit fixes a legacy/mixup of $smime_default_key, I've stumpled while coding the rest.  Since you flatten your history anyway, I refrained from providing a separate PR.

2) Add a new expando, `%i` to `$pgp_entry_format` which expands to the full fingerprint of a key (or its long id, if there is no fingerprint).

3) Discussion worthy:  change the meaning of `%k` and `%i` expandos.  This naming aligns better with the actual names `i` for id and `k` for `key fingerprint`.  Obviously, this is a user visible change, its strictly providing more information as the keyid is a substring of the fingerprint.  I guess almost nobody will notice.  To mitigate the change even further we could change the default `$pgp_entry_format` to use  `%i` instead of `%k`, so only those who changed pgp_entry_format will notice a difference.

*Why this change*: It is well known that short key ids are broken.  Long key ids (double the digits) are a little better but still do not provide the guarantee of no collision. Therefore, the key fingerprint is the only safe way to guarantee that we are dealing with the key we think we are dealing.

### CODE REVIEW

File: docs/config.c [hunk 1]
```diff
@@ -3113,7 +3113,8 @@
 ** .dt %a     .dd Algorithm
 ** .dt %c     .dd Capabilities
 ** .dt %f     .dd Flags
-** .dt %k     .dd Key id
+** .dt %i     .dd Key id
+** .dt %k     .dd Key fingerprint (or long key id if non-existent)
```
Comment (gahr) [inline docs/config.c:3117]: I would not change existing expandos

**Comment by gahr** [comment]
How would you feel about keeping `k` as-is and adding `K`? At the end of the day, aren't you adding a "bigger" version of it? :)

**Comment by rayfordshire** [comment]
> An alternative, might be to use a two-letter expando, %kf (key-fingerprint)?

But how would you print then the key id followed by a literal `f`, e.g. `0x12345f` you cant use `%` to escape `f`?

**Comment by flatcap** [comment]
> But how would you print then the key id followed by a literal `f`, e.g. `0x12345f` (you can't use `%` to escape `f`)?

You wouldn't be able to.
Fortunately, this situation probably isn't that likely.
(the same with the index expandos)

**Comment by rayfordshire** [comment]
> >  But how would you print then the key id followed by a literal f, e.g. 0x12345f (you can't use % to escape f)?
>
>You wouldn't be able to.
>Fortunately, this situation probably isn't that likely.
>(the same with the index expandos)

Just because it's not likely, should we actively irreversible block this possibility?
The expandos for index are now history, so that ship has sailed but here we talk about a new one in a different menu which does not yet have this flaw.

---

I can see why you feel uncomfortable with the swap of `%i` and `%k`, so feel free to omit that commit (that's way it is a separate one).  If you don't like the choice of `%i` feel free to pick another letter.

**Comment by flatcap** [comment]
> should we actively irreversible block this possibility?

No :-)

> If you don't like the choice of `%i` feel free to pick another letter.

How about `%p` for finger**p**rint?

---

In the long-term (I have a lot of plans in this category), it'd be nice to eliminate the hard-to-remember single letters.
One possibility is the form `%{fingerprint}`, discussed briefly here:

https://github.com/neomutt/flatcap-brain-dump/blob/main/split-expando.txt#L100-L114

It'd be quite a bit of work to implement, though.

**Comment by flatcap** [comment]
# Welcome to NeoMutt

Welcome aboard, @rayfordshire!
I don't have an email address for you, so here's my usual [Welcome Letter](https://gist.github.com/flatcap/b589998642e8a9b494e0a15af045ad29).

**Comment by rayfordshire** [comment]
>> If you don't like the choice of %i feel free to pick another letter.
>
> How about `%p` for fingerprint?

Sorry, already taken by protocol.

**Comment by flatcap** [comment]
> Sorry, already taken by protocol.

hehe, OK.
Pick at a letter that isn't in use and we'll go with that.
My opinion doesn't matter :-)

**Comment by rayfordshire** [comment]
> Pick at a letter that isn't in use and we'll go with that.

I already did, it's `%i`

You have just to decide whether you this with or without swapping `%i` and `%k`, which would result in slightly longer "ids" by default.

**Comment by flatcap** [comment]
> with or without swapping `%i` and `%k`

Without swapping, please.

We'll do our best to inform the users, when we make a release.

**Comment by rayfordshire** [comment]
dropped the last commit.

**Review by flatcap — APPROVED** [review]
That's great.  Thanks.
Sorry for all the to-and-fro.

**Pull request merged**